test/parallel/test-native-certs.mjs

```diff
@@ -56,22 +82,75 @@ const handleRequest = (req, res) => {
 };
 
 describe('use-system-ca', function() {
-  let server;
 
-  beforeEach(async function() {
-    server = https.createServer({
-      key: fixtures.readKey('agent8-key.pem'),
-      cert: fixtures.readKey('agent8-cert.pem'),
+  async function setupServer(key, cert) {
```

Comment:
Instead of appending the test case to an existing file, can you put the different tests in different files instead, so that they are easier to debug?

Also we can just put them in a separate test folder. You can do it by

```
mkdir test/system-ca
cp test/module-hooks/testcfg.py test/system-ca/test.cfg.py
cp test/module-hooks/module-hooks.status test/system-ca/system-ca.status
```

Then replace `module-hooks` to `system-ca` in the copied files, and add `system-ca` to the `IGNORED_SUITES` in `tools/test.py` so that it's skipped in the CI. We can then put the instructions on how to add the certificates in `test/system-ca/README.md` instead.